### PR TITLE
[ARCHETYPE-574] Use default values in integration tests goal

### PR DESCRIPTION
**Pull request by Giovds**
1. Add failing test case to the integration tests
2. Use the same configurator as done in the `generate` goal to provide default values.

I had to change some existing integration tests, because apparently it is possible to add non-existing properties to an archetype in the `integration-test` goal, while this is not possible when using the `generate` goal. I've tried to add non-existing properties through the command line; both with `-DsomeProp=test` and through a properties file `-Darchetype.properties=./archetype.properties`. 

In both cases I was unable to include them while generating an archetype, which leads me to believe the integration test adding those properties to the archetype configuration is wrong compared to the actual implementation. It now matches the replacing of properties correctly. Hence the changes in the existing integration tests. WDYT?

---

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [x] Make sure there is a [ARCHETYPE-574](https://issues.apache.org/jira/browse/ARCHETYPE-574). filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[ARCHETYPE-XXX] SUMMARY`,
       where you replace `ARCHETYPE-XXX` and `SUMMARY` with the appropriate JIRA issue.
 - [x] Also format the first line of the commit message like `[ARCHETYPE-XXX] SUMMARY`.
       Best practice is to use the JIRA issue title in both the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by Giovds** [comment]
@slawekjaranowski I'm not sure if you are the right person to ask, but I see you have been working on the plugin recently. What do you think of the change regarding the integration tests (see description of this PR)?

>I had to change some existing integration tests, because apparently it is possible to add non-existing properties to an archetype in the integration-test goal, while this is not possible when using the generate goal. I've tried to add non-existing properties through the command line; both with -DsomeProp=test and through a properties file -Darchetype.properties=./archetype.properties.
<br/> In both cases I was unable to include them while generating an archetype, which leads me to believe the integration test adding those properties to the archetype configuration is wrong compared to the actual implementation. It now matches the replacing of properties correctly. Hence the changes in the existing integration tests. WDYT?

File: maven-archetype-plugin/src/it/projects/ARCHETYPE-574_default-value-should-be-used-in-integration-test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                        <maven.compiler.target>${maven.compiler.target}</maven.compiler.target>
+                    </properties>
+                </configuration>
+            </plugin>
+        </plugins>
+        <extensions>
+            <extension>
+                <groupId>org.apache.maven.archetype</groupId>
+                <artifactId>archetype-packaging</artifactId>
+                <version>3.1.1</version>
```
Comment (slawekjaranowski) [inline maven-archetype-plugin/src/it/projects/ARCHETYPE-574_default-value-should-be-used-in-integration-test/pom.xml:52]: a `@project.version@` should be used

File: maven-archetype-plugin/src/it/projects/ARCHETYPE-574_default-value-should-be-used-in-integration-test/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  <fileSets>
+    <fileSet encoding="UTF-8" filtered="true" packaged="true">
+        <directory>src/main/java</directory>
+        <includes>
+          <include>**/*.java</include>
+        </includes>
+    </fileSet>
+  </fileSets>
+</archetype-descriptor>
```
Comment (slawekjaranowski) [inline maven-archetype-plugin/src/it/projects/ARCHETYPE-574_default-value-should-be-used-in-integration-test/src/main/resources/META-INF/maven/archetype-metadata.xml:38]: missing new line - here and in other place

**Comment by Giovds** [comment]
Still unsure why test fails on agent, but succeeds locally..

**Comment by slawekjaranowski** [comment]
> Still unsure why test fails on agent, but succeeds locally..

I will look at it

**Comment by slawekjaranowski** [comment]
@Giovds build fixed in #235

**Comment by jira-importer** [comment]
Resolve #826

**Pull request merged**